### PR TITLE
Upgrade xml2js to v0.5.0 + migration to @vscode/vsce + migration to @azure/storage-blob

### PR DESCRIPTION
**Pull request by cheenamalhotra**
Consuming dependencies:
- [x] https://www.npmjs.com/package/@azure/ms-rest-js (Updated to [v2.6.6](https://www.npmjs.com/package/@azure/ms-rest-js/v/2.6.6))
- [x] https://www.npmjs.com/package/azure-storage > Migrated to [@azure/storage-blob](https://www.npmjs.com/package/@azure/storage-blob)
- [x] https://www.npmjs.com/package/@azure/core-http (Updated to [v3.0.1](https://www.npmjs.com/package/@azure/core-http/v/3.0.1))
  - [x] Updated @azure/storage-blob to [v12.13.0](https://www.npmjs.com/package/@azure/storage-blob/v/12.13.0)
 - [x] https://www.npmjs.com/package/vsce  > Migrated to [@vscode/vsce](https://www.npmjs.com/package/@vscode/vsce) v2.16.0
   - With this update, few more changes were made, that I'll summarize:
     - I've updated `extensions.ts` with code from VSCode reporsitory in code paths that are not SQL Carbon edited, to make sure we're calling the APIs in the right way.
     - Extension output directories e.g. `.build\external\extensions\***` don't contain `yarn.lock` and therefore when calling `vsce.createVSIX`, `useYarn` is set to `false` to fix the error during packaging extensions ([error log](https://mssqltools.visualstudio.com/CrossPlatBuildScripts/_build/results?buildId=196712&view=logs&j=1c20ec05-7612-5293-9b87-7ea15c22422f&t=f869d998-330e-5751-caff-acb8d65e4e21&l=282))
        ```log
        [17:36:34] 'package-external-extensions' errored after 5.87 min
        [17:36:34] Error: Command failed: yarn list --prod --json
        {"type":"warning","data":"package.json: License should be a valid SPDX license expression"}
        {"type":"error","data":"No lockfile in this directory. Run `yarn install` to generate one."}
        ```
   - ~https://www.npmjs.com/package/@vscode/vsce (_update not available_ - Not tracked as of now, as it requires additional changes reverted in [commit](https://github.com/microsoft/azuredatastudio/pull/22664/commits/e165f0ca88723c82218db4ba3201456e70f56a69))~

### CODE REVIEW

**Review by cssuh — APPROVED** [review]

File: extensions/machine-learning/package.json [hunk 1]
```diff
@@ -155,6 +155,9 @@
     "lodash": "^4.17.21",
     "@microsoft/azdata-test": "^3.0.1"
   },
+  "resolutions": {
+    "xml2js": "^0.5.0"
```
Comment (Charles-Gagnon) [inline extensions/machine-learning/package.json:159]: Did you look at whether the azure packages that use this are being updated? That would be greatly preferred over using resolutions.

Comment (cheenamalhotra) [reply extensions/machine-learning/package.json:159]: The v0.5.0 was released yesterday so I don't see all updates are available yet, I've listed packages to track better in PR description.

File: extensions/csharp/package.json [hunk 2]
```diff
@@ -6,7 +6,7 @@
   "publisher": "vscode",
   "license": "MIT",
   "engines": {
-    "vscode": "0.10.x"
+    "vscode": "^0.10.0"
```
Comment (Charles-Gagnon) [inline extensions/csharp/package.json:9]: We should avoid making updates to VS Code stuff unless necessary - please revert the changes to extensions coming from them

Comment (Charles-Gagnon) [inline extensions/csharp/package.json:9]: We should avoid making updates to VS Code stuff unless necessary - please revert the changes to extensions coming from them

**Comment by Charles-Gagnon** [comment]
Are you done making changes? If not, could you put this in a draft state until it's ready for review?

File: extensions/csharp/package.json (hunk 2 above)
Comment (cheenamalhotra) [reply extensions/csharp/package.json:9]: They are all necessary changes that I've made so far as `@vscode/vsce` now scans extensions better and it's catching all potential issues.

Comment (Charles-Gagnon) [reply extensions/csharp/package.json:9]: Yes, but unless it's a security concern (which for build stuff it shouldn't be) then for the sake of making merges easier it's preferable to wait until VS Code updates their stuff.

Comment (cheenamalhotra) [reply extensions/csharp/package.json]: Ok, downgrading to v2.16.0 for now.

File: build/lib/extensions.ts [hunk 3]
```diff
@@ -74,7 +73,7 @@ export function fromLocal(extensionPath: string, forWeb: boolean): Stream { // {
 			delete data.dependencies;
 			delete data.devDependencies;
 			if (data.main) {
-				data.main = data.main.replace('/out/', /dist/);
+				data.main = data.main.replace('/out/', '/dist/');
```
Comment (Charles-Gagnon) [inline build/lib/extensions.ts:76]: Just to confirm - you took this directly from VS Code? If there's any changes you made yourself then make sure to add SQL CARBON EDIT tags.

File: extensions/julia/package.json [hunk 4]
```diff
@@ -6,7 +6,7 @@
 	"publisher": "vscode",
 	"license": "MIT",
 	"engines": {
-		"vscode": "0.10.x"
+    "vscode": "0.10.x"
```
Comment (Charles-Gagnon) [inline extensions/julia/package.json:9]: undo

**Review by Charles-Gagnon — APPROVED** [review]
Please revert unnecessary changes to VS Code stuff, otherwise looks good. Thanks for taking the time to clean this up!

File: build/lib/extensions.ts (hunk 3 above)
Comment (cheenamalhotra) [reply build/lib/extensions.ts:76]: Sort of... I compared the files and took the changes in that were not within SQL Carbon Edit tags.

File: extensions/dart/package.json [hunk 5]
```diff
@@ -6,7 +6,7 @@
 	"publisher": "vscode",
 	"license": "MIT",
 	"engines": {
-		"vscode": "0.10.x"
+    "vscode": "0.10.x"
```
[the remark this answers is not in the document]
Comment (cheenamalhotra) [reply extensions/dart/package.json:9]: It was using tabs that's why looks strange. 👀 
Need to fix the file to use spaces instead.
